flycheck.el

```diff
@@ -4609,6 +4609,28 @@ symbols in the command."
          (seq-map (lambda (arg) (flycheck-substitute-argument arg checker))
                   (flycheck-checker-arguments checker))))
 
+(defun flycheck--process-send-buffer (process)
+  "Send contents of current buffer to PROCESS.
+This is a compatibility wrapper around `process-send-region`: on
+all systems but Windows it simply calls that function.  On
+Windows, it attempts to work around an Emacs bug by sending the
+contents of the buffer in small batches (4096 bytes seems to be
+the threshold above which Emacs just hangs).
+
+See https://debbugs.gnu.org/cgi/bugreport.cgi?bug=22344 for more
+information about the Emacs issue, and see the original issue at
+https://github.com/flycheck/flycheck/issues/794 for information
+about the impact on Flycheck."
+  (save-restriction
+    (widen)
+    (if (eq system-type 'windows-nt)
+        (let ((from (point-min)))
```

Comment:
I think we should move all of this into a separate function, e.g. `flycheck--process-send-buffer-contents-chunked`.  The function name will be easier to understand than the code itself; it'll help us to understand what it's for in the future.

Furthermore, I think we should move the `system-type` test out of this function into a `defvar`, e.g. `flycheck-chunked-process-input` or so, that holds the value of the `system-type` test.  This lets users easily switch between both, which will help us to test the code on systems other than Windows (e.g. to fix a bug therein), and users to switch if we ask them to, e.g. to check whether their version of Emacs or Node.js is still affected by the bug.

What do you think?